### PR TITLE
[JA] Proofreading on GitHub Actions

### PR DESCRIPTION
**Pull request by sfujiwara**
@lamberta 
Japanese reviewers: @ohtaman, @masa-ita, @AseiSugiyama, @yukinagae, @nuka137, @chie8842, @kiszk

This PR is a proposal of #144 

This PR allows us to automatically apply proofreading tool on GitHub Actions.
It would be a great help for Japanese document reviewers and translators.

I'm happy to hear your opinion about this PR!

## Files and directories

**`.github/workflows/ja.yaml`**

This workflow runs only when directories and files below are changed:

```yaml
on:
  push:
    paths:
      - ".github/workflows/ja.yaml"
      - "site/ja/**"
      - "tools/ja/**"
```

**`tools/ja/`**

This is a proofreading tool used in Japanese community.

## Resulting outputs

We can see the result of proofreading on GitHub PR.
The CI would fail if there are too many errors:

image

We can find the detailed results in GitHub Actions log:

image

### CODE REVIEW

**Comment by lamberta** [comment]
Nice! I'll have a look through this week and will test some things.

File: .github/workflows/ja.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+      - name: Download RedPen
+        run: |
+          wget https://github.com/redpen-cc/redpen/releases/download/redpen-${REDPEN_VERSION}/redpen-${REDPEN_VERSION}.tar.gz
+          tar xvf redpen-${REDPEN_VERSION}.tar.gz
+
+      - name: Set up Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: "3.7"
```
Comment (yashk2810) [inline .github/workflows/ja.yaml:43]: Make the python version 3.6.

Google is at this version so we can;t test anything beyond 3.6.

Comment (sfujiwara) [reply .github/workflows/ja.yaml]: OK. I fixed it!

**Comment by lamberta** [comment]
Hi @sfujiwara , do you know if redpen has a list of supported languages somewhere? I recall it not working for Korean, are there any other restrictions?

You are using this to maintain a translation glossary, yes? (I'm just wondering if other communities can use this setup, as well)

**Comment by kiszk** [comment]
@lamberta [This PR](https://github.com/redpen-cc/redpen/commit/53c0ec434597754d0b41b12be894e988bf0ee71b) intentionally (and temporary) drop supporting Korean due to CI failure. There is [an issue](https://github.com/redpen-cc/redpen/issues/875) to support Korean again. But, it have not been closed yet.

**Comment by sfujiwara** [comment]
@lamberta RedPen has various kinds of validators for example sentence length, invalid words, and so on. Some validators does not support Korean. But, other validators supports all languages including Korean.

I guess RedPen and GitHub Actions are useful for other communities!
See the official document for details:
http://redpen.cc/docs/1.10/index.html#validator

**Comment by sfujiwara** [comment]
Hi, how is it going on this PR?

File: tools/ja/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  - We use RedPen to check markdown files
+- [Jupyter](https://jupyter.org/)
+  - We use Jupyter to convert notebooks to markdowns
+
+## Basic usage
+
+You can run the proofreading tool as below:
+
+```shell script
+./tools/ja/bin/run.sh <target directory or file>
```
Comment (lamberta) [inline tools/ja/README.md:19]:
```suggestion
./tools/ja/bin/proofreading.sh <target directory or file>
```

File: tools/ja/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+```
+
+For example:
+
+```shell script
+./tools/ja/bin/proofreading.sh site/ja/guide
```
Comment (lamberta) [inline tools/ja/README.md:25]: Tested this locally and seems to work ... I used homebrew to install the redpen package, probably out of scope to provide detailed install instructions, but maybe mention the Java dependency---speaking of Java, will this run in the GitHub Actions VM?

There was a log of noise at the command-line when running this, is that usual? Any way to filter out all these deprecation warnings?

**Comment by lamberta** [comment]
Thanks for the bump on this @sfujiwara (been swamped in email lately).

Tested locally and over here: https://github.com/lamberta/tf-docs-l10n/pull/6
and seems to work as advertised.

As long as this is restricted to the ja/ translations then seems fine. Long-term, I'll need your help to maintain this and update if there are structural changes to the docs (things move around). If that's ok with you, then we can merge.

**Comment by sfujiwara** [comment]
@lamberta OK. We will maintain and improve the proofreading tool to make our translation tasks efficient!

File: tools/ja/README.md (hunk 2 above)
Comment (sfujiwara) [reply tools/ja/README.md]: Thanks. I fixed it.

**Comment by sfujiwara** [comment]
Hi @lamberta 

> There was a log of noise at the command-line when running this, is that usual? Any way to filter out all these deprecation warnings?

I added a minor change for the above proposal.

- Separate RedPen logs to files and make it available as GitHub Actions artifacts
- Add `logback.xml` to set RedPen log level `Warn` on GitHub Actions

Merge_pull_request__2_from_sfujiwara_fix-logs_·_sfujiwara_docs-l10n_3c6aca9


> speaking of Java, will this run in the GitHub Actions VM?

Currently it works on my repository and GitHub Actions (I will delete this repository if this PR is merged):
https://github.com/sfujiwara/docs-l10n/actions/runs/107112341

**Comment by lamberta** [comment]
Thanks @sfujiwara for the updates.

@MarkDaoust and @yashk2810 if there's no objections, I'll merge this for the ja folks. If there are maintenance concerns or changes needed, we can loop in @sfujiwara

**Comment by MarkDaoust** [comment]
LGTM.

**Pull request merged**